src/ORM/Behavior/TranslateBehavior.php

```diff
@@ -167,11 +160,16 @@ public function setupFieldAssociations($fields, $table, $fieldConditions, $strat
             ]);
         }
 
+        $conditions = ["$targetAlias.model" => $model];
+        if (!$this->_config['allowEmptyTranslations']) {
+            $conditions["$targetAlias.content !="] = '';
+        }
```

Comment:
That sounds like misinformation to me =). Which database does not treat `<>` and `!=` as equivalent?

Mysql 5.6 does for example:

```
mysql> create table example ( title varchar(10), thing varchar(10) default null );
Query OK, 0 rows affected (0.02 sec)

mysql> insert into example values('null', null), ('empty', ''), ('stuff', 'stuff');
Query OK, 3 rows affected (0.00 sec)
Records: 3  Duplicates: 0  Warnings: 0

mysql> select * from example where thing <> '';
+-------+-------+
| title | thing |
+-------+-------+
| stuff | stuff |
+-------+-------+
1 row in set (0.00 sec)

mysql> select * from example where thing != '';
+-------+-------+
| title | thing |
+-------+-------+
| stuff | stuff |
+-------+-------+
1 row in set (0.00 sec)

mysql>
```

in any event, null translation records are ignored [by the row mapper](https://github.com/cakephp/cakephp/blob/3.0-translate-allow-empty/src/ORM/Behavior/TranslateBehavior.php#L389)